### PR TITLE
Add filter for UUID values

### PR DESCRIPTION
**Pull request by chadicus**
#### What does this PR do?
This pull request adds a string filter for UUID values

#### Checklist
- [ ] Pull request contains a clear definition of changes
- [ ] Tests (either unit, integration, or acceptance) written and passing
- [ ] Relevant documentation produced and/or updated

### CODE REVIEW

File: src/Filter/UuidFilter.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            return null;
+        }
+
+        self::validateVersions($versions);
+        foreach ($versions as $version) {
+            $pattern = sprintf(self::UUID_PATTERN_FORMAT, $version);
+            if (preg_match("/{$pattern}/i", $value)) {
+                return $value;
+            }
+        }
```
Comment (jncarver) [inline src/Filter/UuidFilter.php:84]: Should this support [Nil UUIDs](https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.7)? The `[%d]` and `[89AB]` constraints in the pattern currently prevent them from being used.

Comment (chrisryan) [reply src/Filter/UuidFilter.php:84]: That's a thing?! It's a good question. Would the allow null option control if this was considered valid?

Comment (chadicus) [reply src/Filter/UuidFilter.php:84]: I've updated the filter to allow nil UUIDs. I'm not sure if i like the implementation, but i'd like to get feedback

Comment (chrisryan) [reply src/Filter/UuidFilter.php:84]: The updates look like an appropriate solution. Not sure I have any suggestion to improve the implementation.

Comment (chadicus) [reply src/Filter/UuidFilter.php:84]: It's the best thing i could come up with. I'll add some examples to the README

Comment (chadicus) [reply src/Filter/UuidFilter.php:84]: if approved i will also make the changes to the v3 PR

**Review by jncarver — APPROVED** [review]

File: src/Filter/UuidFilter.php (hunk 1 above)
Comment (jncarver) [reply src/Filter/UuidFilter.php:84]: The updated code looks good to me. Handling `null` values and nil UUIDs separately gives more flexibility to the filter than combining them together in some way.

**Pull request merged**